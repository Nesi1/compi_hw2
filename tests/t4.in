void main(){
    if(ba){
        if(a){
            x=3;
        }
        else{
            x=4;
        }
    }
}

//if (a) if (b) x=3; else x=4;