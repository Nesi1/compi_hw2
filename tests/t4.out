2: Funcs -> epsilon
6: RetType ->  VOID
7: Formals -> epsilon
36: Exp -> ID
36: Exp -> ID
38: Exp -> NUM
18: Statement -> ID ASSIGN Exp SC
12: Statements -> Statement
14: Statement -> LBRACE Statements RBRACE
38: Exp -> NUM
18: Statement -> ID ASSIGN Exp SC
12: Statements -> Statement
14: Statement -> LBRACE Statements RBRACE
23: Statement -> IF LPAREN Exp RPAREN Statement ELSE Statement
12: Statements -> Statement
14: Statement -> LBRACE Statements RBRACE
22: Statement -> IF LPAREN Exp RPAREN Statement
12: Statements -> Statement
4: FuncDecl -> RetType ID LPAREN Formals RPAREN LBRACE Statements RBRACE
3: Funcs -> FuncDecl Funcs
1: Program -> Funcs
